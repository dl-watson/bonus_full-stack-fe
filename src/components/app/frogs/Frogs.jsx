import React from "react";
import PropTypes from "prop-types";

const Frogs = () => {
  return <>Frogs</>;
};

Frogs.propTypes = {};

export default Frogs;
