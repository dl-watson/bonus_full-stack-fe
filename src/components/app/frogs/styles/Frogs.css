* {
    padding: 0;
    margin: 0;
    box-sizing: border-box;
}

.Frogs {
    display: flex;
    flex-direction: column;
    align-items: flex-end;
    width: 75vw;
    
    label {
        text-align: right;
    }

    input {
        width: 300px;
        margin: 5px;
    }

    button {
        width: 100px;
        margin: 5px;
    }
}