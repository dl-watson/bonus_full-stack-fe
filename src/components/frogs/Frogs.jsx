/* eslint-disable max-len */
import React from "react";
import PropTypes from "prop-types";
import styles from "./styles/Frogs.css";

const Frogs = ({ name, image, description, handleSubmit }) => {
  return (
    <form className={styles.Frogs} onSubmit={handleSubmit}>
      <label htmlFor="fname">
        Name:
        <input
          defaultValue={name}
          name="fname"
          id="fname"
          placeholder="Desert rain frog"
        />
      </label>
      <label htmlFor="fimage">
        Image:
        <input
          defaultValue={image}
          name="fimage"
          id="fimage"
          placeholder="https://bit.ly/3jSq7r9"
        />
      </label>
      <label htmlFor="fdescription">
        Description:
        <input
          defaultValue={description}
          name="fdescription"
          id="fdescription"
          placeholder="A species of frog in the family Brevicipitidae."
        />
      </label>
      <button>Submit</button>
    </form>
  );
};

Frogs.propTypes = {
  name: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  handleSubmit: PropTypes.func.isRequired,
};

export default Frogs;
