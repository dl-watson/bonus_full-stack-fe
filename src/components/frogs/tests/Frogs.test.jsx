import React from "react";
import { render, cleanup } from "@testing-library/react";
import Frogs from "../Frogs";

describe("Frogs component", () => {
  afterEach(() => cleanup());
  it("renders Frogs", () => {
    const { asFragment } = render(
      <Frogs name={""} image={""} description={""} handleSubmit={jest.fn()} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
