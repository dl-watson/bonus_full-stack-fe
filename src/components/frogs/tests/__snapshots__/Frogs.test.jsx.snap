// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Frogs component renders Frogs 1`] = `
<DocumentFragment>
  <form
    class="Frogs"
  >
    <label
      for="fname"
    >
      Name:
      <input
        id="fname"
        name="fname"
        placeholder="Desert rain frog"
        value=""
      />
    </label>
    <label
      for="fimage"
    >
      Image:
      <input
        id="fimage"
        name="fimage"
        placeholder="https://bit.ly/3jSq7r9"
        value=""
      />
    </label>
    <label
      for="fdescription"
    >
      Description:
      <input
        id="fdescription"
        name="fdescription"
        placeholder="A species of frog in the family Brevicipitidae."
        value=""
      />
    </label>
    <button>
      Submit
    </button>
  </form>
</DocumentFragment>
`;
